File: client/src/app/tote/tote.ts
export interface ToteEntry {
  internalID: string;
  quantity: number;
}

export interface Tote {
  toteBarcode: string;
  name?: string;
  contents: ToteEntry[];
  notes?: string;
}

export interface ToteEntryView extends ToteEntry {
  internalID: string;
  quantity: number;
  item?: string;
  brand?: string;
  color?: string;
  size?: string;
  type?: string;
  material?: string;
  description?: string;
}

export interface ToteQuantityChangeRequest {
  internalID: string;
  quantity: number;
}

export interface MoveToteItemRequest {
  formToteBarcode: string;
  toToteBarcode: string;
  internalID: string;
  quantity: number;
}

File: server/src/main/java/umm3601/Tote/Tote.java
package umm3601.Tote;

import java.util.ArrayList;
import java.util.List;

import org.mongojack.Id;
import org.mongojack.ObjectId;

@SuppressWarnings({"VisibilityModifier"})
public class Tote {
  @ObjectId @Id
  @SuppressWarnings({"MemberName"})
  public String _id;

  public String toteBarcode;
  public String name;
  public String notes;
  public List<ToteEntry> contents = new ArrayList<>();
}
